src/NUnitFramework/framework/Internal/Execution/WorkItemQueue.cs

```diff
@@ -117,22 +138,29 @@ public bool IsEmpty
         /// <param name="work">The WorkItem to process</param>
         public void Enqueue(WorkItem work)
         {
-            lock (_syncRoot)
+            do
```

Comment:
Spinning is only used in Dequeue, and only for a maximum of five  tries before the thread waits on the event.  This avoids any transitions to the underlying OS kernel for any transitional or short lived usage of the class by another thread.  This means that if another thread is simple using Dequeue to get an entry, in most cases no transition to the OS kernel will occur.  Only if the queue stays empty, or the state stays paused, will the event be waited on.  Using a lock means that a transition to the kernel will occur every time the method is called/exited, even if no other thread is using the class/method.